src/data/set/basic.lean

```diff
@@ -1912,6 +1910,35 @@ subsingleton_empty.pairwise_on r
   pairwise_on {a} r :=
 subsingleton_singleton.pairwise_on r
 
+/-- For a nonempty set `s`, a function `f` takes pairwise equal values on `s` if and only if
+for some `z` in the codomain, `f` takes value `z` on all `x ∈ s`. See also
+`set.pairwise_on_eq_iff_exists_eq` for a version that assumes `[nonempty β]` instead of
+`set.nonempty s`. -/
+theorem nonempty.pairwise_on_eq_iff_exists_eq {s : set α} (hs : s.nonempty) {f : α → β} :
+  (pairwise_on s (λ x y, f x = f y)) ↔ ∃ z, ∀ x ∈ s, f x = z :=
```

Comment:
This generalizes from `(=)` to any equivalence relation, right? Am I right in thinking we have a typeclass for that?

Reply:
Done.